src/main/java/io/confluent/connect/jdbc/source/SbdBulkTableQuerier.java

```diff
@@ -0,0 +1,250 @@
+   * Value is always FALSE, except for the last record where value will be TRUE.
+   */
+  public static final String HEADER_BATCH_COMPLETED = "sbd.batch.completed";
+
+  /**
+   * Index of the record in the batch.
+   * First item has value=1, last item has value=HEADER_BATCH_SIZE
+   */
+  public static final String HEADER_BATCH_INDEX = "sbd.batch.index";
+
+  /**
+   * RFC 3339 datetime when the batch started.
+   * At the first attempt, HEADER_BATCH_STARTED_AT equals HEADER_BATCH_TIME
+   */
+  public static final String HEADER_BATCH_STARTED_AT = "sbd.batch.started.at";
+
+  /**
+   * RFC 3339 datetime when the batch completed.
+   */
+  public static final String HEADER_BATCH_COMPLETED_AT = "sbd.batch.completed.at";
+
+  private static final String THREAD_NAME_PREFIX = "task-thread-";
+  private static final Logger log = LoggerFactory.getLogger(SbdBulkTableQuerier.class);
+
+  private final BatchMetrics metrics;
+  private String currentBatchId;
+  private Instant currentBatchTime;
+  private Instant currentBatchStartTime;
+  private Time time;
+  private int batchSize;
+
+  public SbdBulkTableQuerier(
+      DatabaseDialect dialect,
+      QueryMode mode,
+      String nameOrQuery,
+      String topicPrefix,
+      String suffix,
+      Map<String, Object> offset,
+      Time time
+  ) {
+    super(dialect, mode, nameOrQuery, topicPrefix, suffix);
+
+    // register SBD MBeans
+    String taskId = "undefined";
+    if (Thread.currentThread().getName().startsWith(THREAD_NAME_PREFIX)) {
+      taskId = Thread.currentThread().getName().substring(THREAD_NAME_PREFIX.length());
+    }
+    metrics = new BatchMetrics(taskId, tableId != null ? tableId.tableName() : "query");
+    this.time = time;
+
+    currentBatchId = null;
+    currentBatchTime = null;
+    if (offset != null && offset.containsKey(HEADER_BATCH_ID)) {
+      String batchId = (String) offset.get(HEADER_BATCH_ID);
+      boolean batchCompleted = (boolean) offset.get(HEADER_BATCH_COMPLETED);
+      Instant batchTime = Instant.parse((String) offset.get(HEADER_BATCH_TIME));
+
+      if (batchCompleted) {
+        lastUpdate = batchTime.toEpochMilli();
+      } else {
+        // if the batch is not completed we assume all the previous ones are completed or aborted
+        // so we will only retry the current one if needed
+        currentBatchId = batchId;
+        currentBatchTime = batchTime;
+      }
+    }
+
+    if (currentBatchId != null) {
+      log.info("Configured to retry batch.id={}, batch.time={}", currentBatchId, currentBatchTime);
+    }
+  }
+
+  public long getLastUpdate() {
+    return lastUpdate;
+  }
+
+  protected ResultSet executeQuery() throws SQLException {
+    return stmt.executeQuery();
+  }
+
+  @Override
+  protected void createPreparedStatement(Connection db) throws SQLException {
+    ExpressionBuilder builder = dialect.expressionBuilder();
+    switch (mode) {
+      case TABLE:
+        builder.append("SELECT * FROM ").append(tableId);
```

Comment:
is tableId already escaped or sorrounded by "s if needed?

Reply:
This line comes from the overridden method. Was copy-pasted like this.
tableId is set as follow: `tableId = dialect.parseTableIdentifier(nameOrQuery)`